SimDataFormats/GeneratorProducts/src/HepMCProduct.cc

```diff
@@ -126,16 +125,14 @@ HepMCProduct::HepMCProduct(HepMCProduct const& other) : evt_(nullptr) {
   isVtxGenApplied_ = other.isVtxGenApplied_;
   isVtxBoostApplied_ = other.isVtxBoostApplied_;
   isPBoostApplied_ = other.isPBoostApplied_;
-  //fTimeOffset = other.fTimeOffset;
 }
 
 // swap
 void HepMCProduct::swap(HepMCProduct& other) {
   std::swap(evt_, other.evt_);
-  std::swap(isVtxGenApplied_, other.isVtxGenApplied_);
-  std::swap(isVtxBoostApplied_, other.isVtxBoostApplied_);
-  std::swap(isPBoostApplied_, other.isPBoostApplied_);
-  //std::swap(fTimeOffset, other.fTimeOffset);
+  isVtxGenApplied_ = other.isVtxGenApplied_;
+  isVtxBoostApplied_ = other.isVtxBoostApplied_;
+  isPBoostApplied_ = other.isPBoostApplied_;
```

Comment:
Sorry but strictly speaking this is not a correct implementation of `swap()`, because the values from `this` do not end up in `other` (see https://en.cppreference.com/w/cpp/named_req/Swappable). I'm puzzled though why
https://github.com/cms-sw/cmssw/blob/684a7404ef8464c20f3dc809f798098105bdf63c/DataFormats/Common/interface/Wrapper.h#L89-L94
ends up calling `swap()`.

Ah, it's because the move constructor and assignment are implemented with `swap()`
https://github.com/cms-sw/cmssw/blob/684a7404ef8464c20f3dc809f798098105bdf63c/SimDataFormats/GeneratorProducts/src/HepMCProduct.cc#L146-L150

Maybe the real problem is that the move constructor does not initialize the bools?

Reply:
ah you are right @makortel . Let me revert this change and initialize bools in move constructor

Reply:
thanks @makortel , indeed properly initializing data members for move constructor fixes the issue. I have opened https://github.com/cms-sw/cmssw/pull/47074